alicloud/resource_alicloud_polardb_cluster.go

```diff
@@ -0,0 +1,426 @@
+	"strings"
+	"time"
+
+	"strconv"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/polardb"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+)
+
+func resourceAlicloudPolarDBCluster() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAlicloudPolarDBClusterCreate,
+		Read:   resourceAlicloudPolarDBClusterRead,
+		Update: resourceAlicloudPolarDBClusterUpdate,
+		Delete: resourceAlicloudPolarDBClusterDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(20 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"db_type": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"db_version": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"db_node_class": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"zone_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"cluster_network_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					return true
+				},
+				Deprecated: "Field 'cluster_network_type' has been deprecated from provider version 1.5.0.",
+			},
+			"cluster_charge_type": {
+				Type:         schema.TypeString,
+				ValidateFunc: validation.StringInSlice([]string{string(Postpaid), string(Prepaid)}, false),
+				Optional:     true,
+				Default:      Postpaid,
+			},
+			"renewal_status": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  RenewNormal,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(RenewAutoRenewal),
+					string(RenewNormal),
+					string(RenewNotRenewal)}, false),
+			},
+			"auto_renew": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"auto_renew_period": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      1,
+				ValidateFunc: validation.IntInSlice([]int{1, 2, 3, 6, 12}),
+			},
+			"period": {
+				Type:         schema.TypeInt,
+				ValidateFunc: validation.IntInSlice([]int{1, 2, 3, 4, 5, 6, 7, 8, 9, 12, 24, 36}),
+				Optional:     true,
+				Default:      1,
+			},
+			"security_ips": {
+				Type:     schema.TypeSet,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Computed: true,
+				Optional: true,
+			},
+			"vswitch_id": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Optional: true,
+			},
+			"maintain_time": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"cluster_name": {
```

Comment:
please keep parameter name sync with API. If API does not have ClusterName, also do not use it in the TF.